File: openbis_api/source/java/ch/ethz/sis/openbis/generic/asapi/v3/dto/event/fetchoptions/EventSortOptions.java
package ch.ethz.sis.openbis.generic.asapi.v3.dto.event.fetchoptions;

import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.fetchoptions.SortOptions;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.fetchoptions.SortOrder;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.event.Event;
import ch.systemsx.cisd.base.annotation.JsonObject;
import com.fasterxml.jackson.annotation.JsonIgnore;

@JsonObject("as.dto.event.fetchoptions.EventSortOptions")
public class EventSortOptions extends SortOptions<Event>
{
    @JsonIgnore
    public static final String ID = "event_id";

    @JsonIgnore
    public static final String IDENTIFIER = "event_identifier";

    @JsonIgnore
    public static final String REGISTRATION_DATE = "event_registration_date";

    public SortOrder id()
    {
        return getOrCreateSorting(ID);
    }

    public SortOrder getId()
    {
        return getSorting(ID);
    }

    public SortOrder identifier()
    {
        return getOrCreateSorting(IDENTIFIER);
    }

    public SortOrder getIdentifier()
    {
        return getSorting(IDENTIFIER);
    }

    public SortOrder registrationDate()
    {
        return getOrCreateSorting(REGISTRATION_DATE);
    }

    public SortOrder getRegistrationDate()
    {
        return getSorting(REGISTRATION_DATE);
    }

}

File: openbis/source/java/ch/systemsx/cisd/openbis/public/resources/api/v3/as/dto/event/fetchoptions/EventSortOptions.js
define([ "require", "stjs", "as/dto/common/fetchoptions/SortOptions" ], function(require, stjs, SortOptions) {
	var EventSortOptions = function() {
		SortOptions.call(this);
	};

	var fields = {
	    ID : "event_id",
	    IDENTIFIER : "event_identifier",
        REGISTRATION_DATE : "event_registration_date"
    };

	stjs.extend(EventSortOptions, SortOptions, [ SortOptions ], function(constructor, prototype) {
		prototype['@type'] = 'as.dto.event.fetchoptions.EventSortOptions';
		constructor.serialVersionUID = 1;

        prototype.id = function() {
            return this.getOrCreateSorting(fields.ID);
        };
        prototype.getId = function() {
            return this.getSorting(fields.ID);
        };
        prototype.identifier = function() {
            return this.getOrCreateSorting(fields.IDENTIFIER);
        };
        prototype.getIdentifier = function() {
            return this.getSorting(fields.IDENTIFIER);
        };
        prototype.registrationDate = function() {
            return this.getOrCreateSorting(fields.REGISTRATION_DATE);
        };
        prototype.getRegistrationDate = function() {
            return this.getSorting(fields.REGISTRATION_DATE);
        };
	}, {});
	return EventSortOptions;
})